modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -621,6 +621,8 @@ Object encodeLiteral(Object literal, Class<?> targetType) {
             }
             // by default, return string as is
             return literal;
+        } else if (literal instanceof Number) {
+            return literal;
```

Comment:
Doing an Integer / Double / Long check might be a good idea. 

You could always add another test to see what happens to mongodb when you give it a filter literal containing a BigDecimal or Byte.